src/endpoints/assets/v1.ts

```diff
@@ -0,0 +1,618 @@
+
+    async get_assets(
+        options?: {
+            collection_name?: NameType[]
+            schema_name?: NameType[]
+            template_id?: Int32Type[]
+            burned?: boolean
+            owner?: NameType[]
+            match?: string
+            search?: string
+            match_immutable_name?: string
+            match_mutable_name?: string
+            is_transferable?: boolean
+            is_burnable?: boolean
+            minter?: NameType[]
+            burner?: NameType[]
+            initial_receiver?: NameType[]
+            collection_blacklist?: NameType[]
+            collection_whitelist?: NameType[]
+            only_duplicate_templates?: boolean
+            has_backed_tokens?: boolean
+            authorized_account?: NameType
+            template_whitelist?: Int32Type[]
+            template_blacklist?: Int32Type[]
+            hide_templates_by_accounts?: NameType[]
+            hide_offers?: boolean
+            asset_id?: UInt64Type[]
+            ids?: UInt64Type[]
+            lower_bound?: string
+            upper_bound?: string
+            before?: number
+            after?: number
+            page?: number
+            limit?: number
+            order?: 'asc' | 'desc'
+            sort?: 'asset_id' | 'minted' | 'updated' | 'transferred' | 'template_mint' | 'name'
+        },
+        extra_options?: {[key: string]: string}
+    ) {
+        const queryParts = {}
+
+        for (const [key, value] of Object.entries(options || {})) {
+            queryParts[key] = value as string
+        }
+
+        for (const [key, value] of Object.entries(extra_options || {})) {
+            queryParts[key] = value
+        }
+
+        const queryParams = Object.keys(queryParts).length
+            ? '?' + new URLSearchParams(queryParts).toString()
+            : ''
+
+        return this.client.call({
+            path: `/atomicassets/v1/assets${queryParams}`,
+            method: 'GET',
+            responseType: Assets.GetAssetsResponse,
+        })
+    }
+
+    async get_asset(asset_id: UInt64Type) {
+        return this.client.call({
+            path: `/atomicassets/v1/assets/${asset_id}`,
+            method: 'GET',
+            responseType: Assets.GetAssetResponse,
+        })
+    }
+
+    async get_asset_stats(asset_id: UInt64Type) {
+        return this.client.call({
+            path: `/atomicassets/v1/assets/${asset_id}/stats`,
+            method: 'GET',
+            responseType: Assets.GetAssetStatsResponse,
+        })
+    }
+
+    async get_asset_logs(
+        asset_id: UInt64Type,
+        options?: {
+            page?: number
+            limit?: number
+            order?: 'asc' | 'desc'
+            action_whitelist?: ActionType[]
+            action_blacklist?: ActionType[]
+        }
+    ) {
+        const queryParts = {}
+
+        for (const [key, value] of Object.entries(options || {})) {
+            queryParts[key] = value as string
```

Comment:
Same comment here, maybe we should be doing `queryParts[key] = String(value)` instead?

Reply:
Uh, yeah, just checked that ABISerializable doesn't implement cast methods.